forge-game/src/main/java/forge/game/player/Player.java

```diff
@@ -3116,7 +3153,72 @@ else if (game.getRules().hasAppliedVariant(GameType.Oathbreaker)) {
         eff.addStaticAbility(mayBePlayedAbility);
         return eff;
     }
+    public void createTheRing(Card host) {
+        final PlayerZone com = getZone(ZoneType.Command);
+        if (theRing == null) {
+            theRing = new Card(game.nextCardId(), null, game);
+            theRing.setOwner(this);
+            theRing.setImmutable(true);
+            String image = ImageKeys.getTokenKey("the_ring");
+            if (host != null) {
+                theRing.setImageKey("t:the_ring_" + host.getSetCode().toLowerCase());
+                theRing.setSetCode(host.getSetCode());
+            } else {
+                theRing.setImageKey(image);
+            }
+            theRing.setName("The Ring");
+            theRing.updateStateForView();
+            com.add(theRing);
+            this.updateZoneForView(com);
+        }
+    }
+    public void setRingLevel(int level) {
+        if (getTheRing() == null)
+            createTheRing(null);
+        if (level == 1) {
```

Comment:
One thing what we could do is to make RingLevel as Property of the CardTraits

Then they are only active if the NumRingTemptedYou >= RingLevel

Then we could all all of the card traits when the Ring is created

Reply:
can we just parse the player ConditionCheckSVar$ NumRingTempted (SVar:NumRingTempted:PlayerCountPropertyYou$RingTemptedYou) GE1 level 1 static is active and so on and so forth..?

Reply:
i wanted to do it similar to ClassLevel so its more cleaner

Reply:
Ok I'll just leave this MR and let you or somebody will update the code to much cleaner version

Reply:
@Hanmac Gonna just spin up a new issue for cleanup https://github.com/Card-Forge/forge/issues/3535